src/main/java/com/google/idea/perf/CallTreeBuilder.kt

```diff
@@ -69,11 +70,20 @@ class CallTreeBuilder(
     fun push(tracepoint: Tracepoint) {
         val parent = currentNode
         val child = parent.children.getOrPut(tracepoint) { Tree(tracepoint, parent) }
-        val now = clock.sample()
+        val flags = tracepoint.flags.get()
+
+        child.tracepointFlags = flags
+
+        if ((flags and TracepointFlags.TRACE_CALL_COUNT) != 0) {
+            child.callCount++
```

Comment:
Since incrementing `callCount` is so cheap, one simplification could be to remove the `TRACE_CALL_COUNT` flag and just assume it is always `true`.

(Unless the tracepoint is disabled via an `untrace` command---but in that case I think it might be better to remove the method instrumentation entirely so that it does not call into `Trampoline`. See my comment in `TracerController`.)

If we do end up removing `TRACE_CALL_COUNT`, then we could even consider turning `TRACE_WALL_TIME` into a simple boolean field rather than an offset into a bitset (unless there are other flags we plan on adding?).

What do you think?

Reply:
My thought was that we were going to add memory allocation and GC time tracing and later on, which is why I added a `flags` field. Maybe we can consider turning the field into a boolean, but I'm still not completely sure about it yet.

Reply:
Sounds good to me! I'm OK with either approach.